File: Java_Programmers/1단계/src/두정수사이의합.java
public class 두정수사이의합 {
    public long solution(int a, int b) {
        long answer = 0;
        if(a == b) return a;
        if(b < a){
            int temp = a;
            a = b;
            b = temp;
        }
        for(int i = a; i <= b; i++){
            answer += i;
        }
        return answer;
    }
    public long solution2(int a, int b){
        if(a == b) return a;
        if(b < a){
            int temp = a;
            a = b;
            b = temp;
        }
        return sumAtoB(a, b);
    }
    // 등차수열의 합공식 :  n(a + l) /2
    public long sumAtoB(int a, int b){
        return ((long) (b - a + 1) *(a+b) / 2);
    }

    public static void main(String[] args) {
        두정수사이의합 sol = new 두정수사이의합();
        int a = 3;
        int b = 5;
        System.out.println(sol.solution(a, b));
        System.out.println(sol.solution2(a, b));
    }
}

File: python/python_Programmers/1단계/두정수사이의합.py
# 내 풀이
def solution(a, b):
    answer = 0
    return sum([answer+i for i in range(min(a, b), max(a, b)+1)])

# 내 풀이2
def solution2(a, b):
    return sum(range(min(a, b), max(a, b)+1))

# 모범 풀이
def solution3(a, b):
    if a > b: a, b = b, a
    return sum(range(a,b+1))

# 등차수열 풀이
def solution4(a, b):
    return (abs(a-b)+1)*(a+b)//2
           # 항의 개수 * 첫항,끝항의 합 / 2
a = 5
b = 3
print(solution4(a, b))

